index.js

```diff
@@ -153,8 +168,16 @@ NYC.prototype._wrapRequire = function () {
 
     _this.sourceMapCache.add(filename, code)
 
-    // now instrument the compiled code.
-    return _this.instrumenter.instrumentSync(code, './' + relFile)
+    var hash = md5(code)
+    var cacheFilePath = path.join(_this._cacheDirectory(), hash + '.js')
+
+    try {
+      return fs.readFileSync(cacheFilePath, 'utf8')
+    } catch (e) {
+      var instrumented = _this.instrumenter().instrumentSync(code, './' + relFile)
+      fs.writeFile(cacheFilePath, instrumented)
```

Comment:
I'd assume the writes are consecutive, not parallel, so even if there is a collision it just means the cache file is written a couple times. It'll still be a valid cache file in the end. Errors won't be reported though, e.g. if we somehow screw up writing the cache dir we'd never notice. OTOH if we fail to write the cache file once in a while that's not the end of the world (and a test run shouldn't fail because of it).

I suppose this could use async with a `console.warn() if there are errors?